client/src/general_utils.ts

```diff
@@ -133,7 +133,32 @@ export const retry_promise = (
       );
   });
 };
-
+interface completeAssignInterface {
```

Comment:
I _think_ function overloading can't be done via interfaces, unless that function in a method of an object/class (in the lodash case, it's a method on the `_` object).

This works 
```
export function completeAssign<TObject, TSource>(
  object: TObject,
  source: TSource
): TObject & TSource;
export function completeAssign<TObject, TSource1, TSource2>(
  object: TObject,
  source1: TSource1,
  source2: TSource2
): TObject & TSource1 & TSource2;
export function completeAssign<TObject, TSource1, TSource2, TSource3>(
  object: TObject,
  source1: TSource1,
  source2: TSource2,
  source3: TSource3
): TObject & TSource1 & TSource2 & TSource3;
export function completeAssign<TObject, TSource1, TSource2, TSource3, TSource4>(
  object: TObject,
  source1: TSource1,
  source2: TSource2,
  source3: TSource3,
  source4: TSource4
): TObject & TSource1 & TSource2 & TSource3 & TSource4;
export function completeAssign<TObject>(object: TObject): TObject;
export function completeAssign(target: any, ...sources: any[]): any {
  sources.forEach((source) => {
    let descriptors = Object.keys(source).reduce(
      (descriptors: { [key: string]: PropertyDescriptor }, key) => {
        descriptors[key] = Object.getOwnPropertyDescriptor(source, key)!;
        return descriptors;
      },
      {}
    );
    // by default, Object.assign copies enumerable Symbols too
    Object.getOwnPropertySymbols(source).forEach((sym) => {
      let descriptor = Object.getOwnPropertyDescriptor(source, sym);
      if (descriptor && descriptor.enumerable) {
        // Typescript bug that symbol cannot be used as index: https://github.com/microsoft/TypeScript/issues/1863
        descriptors[(sym as unknown) as string] = descriptor;
      }
    });
    Object.defineProperties(target, descriptors);
  });
  return target;
}
```
**But** we get a bunch of eslint no-redeclare warnings, ugh. Looks like we just need to switch to the typescript-eslint version of the no-redeclare rule though.
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-redeclare.md